kinto/views/permissions.py

```diff
@@ -82,12 +82,14 @@ def get_records(self, filters=None, sorting=None, pagination_rules=None,
         # Check settings for every allowed resources.
         from_settings = allowed_from_settings(self.request.registry.settings, principals)
 
-        # If 'bucket_create_principals' is in settings, setup the '/buckets' uri with
-        # permissions and add the permission lookup
-        bucket_creation = from_settings.get('')
-        if bucket_creation:
-            perms_by_object_uri['/buckets'] = bucket_creation
-            perms_descending_tree[''] = {'bucket:create': {'': {'bucket:create'}}}
+        # Setup additional bucket permissions from settings
+        bucket_perms = from_settings.get('')
+        if bucket_perms:
+            for bucket_perm in bucket_perms:
+                perms_by_object_uri['/buckets'] = {bucket_perm}
+                perms_descending_tree['bucket'].update({
+                    bucket_perm: {'bucket': {bucket_perm}}
+                })
```

Comment:
I found that the accounts plugin also populates `from_settings` with `account:create`, so I added a for loop as well that will allow for multiple settings.

Reply:
Oh! Very good catch!

I think we'll have to do the same thing for accounts, I'm pretty sure the permission to create and write account does not appear in the permission endpoint. We should open an issue and fix it in another PR.